### DISCUSSION TITLE
Getting the state of the picker from a custom action

### DISCUSSION THREAD

**Discussion by kzmath** [nvim-mini/mini.nvim#682] [General]
I am experimenting with mini.pick instead of telescope, and I am enjoying the simplicity so far!

From reading the documentation, it looks like custom action callbacks are called without arguments. The only exception is the "choose" action, which you can have only one per source. Is there a way using the current api to say get the currently picked item in a custom action? The use case I had in mind is to implement some functions of "telescope file_browser".

**Comment by echasnovski** [comment]
If I understood the question correctly, the answer is [`MiniPick.get_picker_matches()`](https://github.com/echasnovski/mini.nvim/blob/79b1b3350f1cac7cef9e46a637f9adf2a14b4923/doc/mini-pick.txt#L1217). In particular field `current` contains the item that is current at the moment of its call.

> The use case I had in mind is to implement some functions of "telescope file_browser".

I'd mention it just in case: there is [`explorer()`](https://github.com/echasnovski/mini.nvim/blob/79b1b3350f1cac7cef9e46a637f9adf2a14b4923/doc/mini-extra.txt#L287) picker in 'mini.extra', which is similar to `Telescope file_browser`.
